pkg/api/types.go

```diff
@@ -1004,6 +1008,30 @@ type Service struct {
 	Status ServiceStatus `json:"status,omitempty"`
 }
 
+// ServiceAccount binds together:
+// * a name, understood by users, and perhaps by peripheral systems, for an identity
+// * a principal that can be authenticated and authorized
+// * a security context, which defines the Linux Capabilities, User IDs, Groups IDs, and other capabilities and controls on interaction with the file system and OS.
+// * a set of secrets
+type ServiceAccount struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// APISecret is the name of the secret to use to identify this ServiceAccount to the API
+	// It must be of type SecretTypeServiceAccountToken
+	APISecret string `json:"apiSecret"`
+	// Secrets is the list of secrets allowed to be used by pods running using this ServiceAccount
+	Secrets []string `json:"secrets"`
```

Comment:
So, apiSecret is for this service account to talk to kubernetes/openshift apiserver, and secret is to talk to other things, like, say, the GCE API or the github API? 

Should we call it KubeSecret?  Or do you not want to leak the k word into OS API (understandable)?  Other ideas?